msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -7334,6 +7496,7 @@ else if (target instanceof Topic){
           mask.put(PermissionLevel.READ, Boolean.valueOf(permBean.getRead()));
           mask.put(PermissionLevel.MARK_AS_READ,Boolean.valueOf(permBean.getMarkAsRead()));
           mask.put(PermissionLevel.MODERATE_POSTINGS, Boolean.valueOf(permBean.getModeratePostings()));
+          mask.put(PermissionLevel.IDENTIFY_ANON_AUTHORS, Boolean.valueOf(permBean.getIdentifyAnonAuthors()));
           mask.put(PermissionLevel.DELETE_OWN, Boolean.valueOf(permBean.getDeleteOwn()));
           mask.put(PermissionLevel.DELETE_ANY, Boolean.valueOf(permBean.getDeleteAny()));
           mask.put(PermissionLevel.REVISE_OWN, Boolean.valueOf(permBean.getReviseOwn()));
```

Comment:
Probably. The if statements containing these blocks might not even be necessary (but I'm not certain cause PermissionLevelImpl overrides .equals with a very incoherent implementation). At the very least a method would be nice here, so I'll set one up.